staging/kos/pkg/controllers/ipam/ipam.go

```diff
@@ -465,11 +469,34 @@ func (ctlr *IPAMController) processNetworkAttachment(ns, name string) error {
 		return nil
 	} else {
 		lockForStatus, ipForStatus, err = ctlr.pickAndLockAddress(ns, name, att, subnetName, desiredVNI, desiredBaseU, desiredLastU)
-		if err != nil {
+		if isFullSubnetErr(err) && !fullSubnetMsgFound(att.Status.Errors.IPAM) {
```

Comment:
This PR is changing subnet-full from something that causes retry to something that determines part of Status.  Regardless of what Status previously contained.  The logic here gets that last part wrong.  We probably want to adopt the pattern of computing what Status should say and then requesting an update if that differs from what Status currently says.

Reply:
> This PR is changing subnet-full from something that causes retry to something that determines part of Status. Regardless of what Status previously contained

Both sentences are false. This PR is changing subnet-full from something that causes retry to something that causes a retry AND determines part of Status. If the subnet is full there will always be a retry, but if the processed NetworkAttachment status already contains an error message describing a full-subnet condition there is no status update.

Reply:
Right.

Reply:
Are you convinced of the correctness of the aforementioned code or is there something wrong you still see in it?

Reply:
The code is good.  You explained the errors in my original comment.